iree/compiler/Dialect/Util/Transforms/FoldGlobals.cpp

```diff
@@ -382,6 +385,8 @@ class FoldGlobalsPass
     FrozenRewritePatternSet frozenPatterns(std::move(patterns));
 
     auto moduleOp = getOperation();
+    beforeFoldingGlobals =
```

Comment:
use llvm::count for these - we don't want to be allocating a new vector and growing it several times just to get the size - especially because this won't be stripped in release builds when statistics are off (as it has side effects)

Reply:
Good point about stripping in builds with statistics being off.

This works, but there may be a simpler helper?

```c++
    afterFoldingGlobals =
        llvm::count_if(moduleOp.getOps<IREE::Util::GlobalOp>(),
                       [](IREE::Util::GlobalOp op) { return true; });
```

Reply:
hmm ADT is always weird about these things - there's an llvm::size, but that may not work here (worth a shot) - otherwise would need to do something like std::distance, which doesn't take ranges, so could have a `count` helper:
```
template <typename R>
static size_t count(R &&range) {
  return std::distance(range.begin(), range.end());
}
```

Reply:
(that should really be in the ADT as an llvm::distance that takes a range, but ehhh)

Reply:
```c++
beforeFoldingGlobals = llvm::size(moduleOp.getOps<IREE::Util::GlobalOp>());
```

--->

`[build] D:\dev\projects\iree\iree\compiler\Dialect\Util\Transforms\FoldGlobals.cpp(394): error C2672: 'llvm::size': no matching overloaded function found
[build] D:\dev\projects\iree\iree\compiler\Dialect\Util\Transforms\FoldGlobals.cpp(394): error C2893: Failed to specialize function template 'auto llvm::size(R &&,enable_if<std::is_base_of<std::random_access_iterator_tag,std::iterator_traits<unknown-type>::iterator_category>::value,void>::type *)'
[build] D:\dev\projects\iree\third_party\llvm-project\llvm\include\llvm/ADT/STLExtras.h(1532): note: see declaration of 'llvm::size'
[build] D:\dev\projects\iree\iree\compiler\Dialect\Util\Transforms\FoldGlobals.cpp(394): note: With the following template arguments:
[build] D:\dev\projects\iree\iree\compiler\Dialect\Util\Transforms\FoldGlobals.cpp(394): note: 'R=llvm::iterator_range<mlir::detail::op_iterator<OpT,mlir::Region::OpIterator>>'`

https://github.com/llvm/llvm-project/blob/4ae8c8310401b665cb2728c1bf68ecd4b10cee78/llvm/include/llvm/ADT/STLExtras.h#L1529-L1539 🤔 

`count` helper worked though :)